src/PersistentPageIdentifiersExtension.php

```diff
@@ -30,19 +31,13 @@ public function getIdGenerator(): IdGenerator {
 	}
 
 	public function getPersistentPageIdentifiersRepo(): PersistentPageIdentifiersRepo {
-		if ( defined( 'MW_PARSER_TEST' ) ) {
-			return $this->getParserTestPersistentPageIdentifiersRepo();
+		if ( $this->persistentPageIdentifiersRepo === null ) {
```

Comment:
Nitpick: the field could be non-nullable (and without initializer) with an `!isset( $this->persistentPageIdentifiersRepo )` check